modules/objc/examples/code_snippets/SampleCodeTest.m

```diff
@@ -1368,1143 +1376,1271 @@ - (void) dontTestListenerCertificateAuthenticatorCallback {
-    public func stop()
-}
+        NSError* error = nil;
+        CBLMutableDictionary* myDict = [[CBLMutableDictionary alloc] initWithJSON: aJSONstring error: &error];
+        NSLog(@"%@", myDict);
 
-// end::p2p-ws-api-urlendpointlistener[]
+        NSString* name = [myDict stringForKey: @"name"];
+        NSLog(@"Details for: %@", name);
 
+        for (NSString* key in myDict) {
+            NSLog(@"%@ %@", key, [myDict valueForKey: key]);
+        }
 
-// tag::p2p-ws-api-urlendpointlistener-constructor[]
-let config = URLEndpointListenerConfiguration.init(database: self.oDB)
-thisConfig.port = tls ? wssPort : wsPort
-thisConfig.disableTLS = !tls
-thisConfig.authenticator = auth
-self.listener = URLEndpointListener.init(config: config) // <1>
+    // end:tojson-dictionary[]
+}
 
-// end::p2p-ws-api-urlendpointlistener-constructor[]
+- (void) dontTestToJSONDocument {
+    // Example 7. Documents as JSON strings
+    // tag::tojson-document[]
+    NSError* error = nil;
+    CBLDatabase* db = [[CBLDatabase alloc] initWithName: @"hotel" error: &error];
+    CBLDatabase* dbnew = [[CBLDatabase alloc] initWithName: @"newhotels" error: &error];
+    
+    CBLQuery* listQuery = [CBLQueryBuilder select: @[[CBLQuerySelectResult expression: [CBLQueryMeta id] as: @"metaId"]]
+                                             from: [CBLQueryDataSource database: db]];
+    
+    
+    CBLQueryResultSet* rs = [listQuery execute: &error];
+    for (CBLQueryResult* r in rs.allObjects) {
+        NSString* thisId = [r stringForKey: @"metaId"];
+        NSString* thisJSONstring = [[db documentWithID: thisId] toJSON];
+        
+        NSLog(@"JSON String = %@", thisJSONstring);
+        
+        NSError* docError = nil;
+        CBLMutableDocument* hotelFromJSON = [[CBLMutableDocument alloc] initWithID: thisId json: thisJSONstring error: &docError];
+        
+        [dbnew saveDocument: hotelFromJSON error: &error];
+        CBLDocument* newhotel = [dbnew documentWithID: thisId];
+        NSArray* keys = newhotel.keys;
+        for (NSString* key in keys) {
+            NSLog(@"%@ %@", key, [newhotel valueForKey: key]);
+        }
+    }
+    // end::tojson-document[]
+}
 
+- (void) dontTestToJSONArray {
+    CBLDatabase* db = self.db;
+    CBLDatabase* newdb = self.db;
+    
+    // tag::tojson-array[]
+    NSString* thisJSONstring = @"[{\"id\":\"1000\",\"type\":\"hotel\",\"name\":\"Hotel Ted\",\"city\":\"Paris\","
+        "\"country\":\"France\",\"description\":\"Undefined description for Hotel Ted\"},"
+        "{\"id\":\"1001\",\"type\":\"hotel\",\"name\":\"Hotel Fred\",\"city\":\"London\","
+        "\"country\":\"England\",\"description\":\"Undefined description for Hotel Fred\"},"
+        "{\"id\":\"1002\",\"type\":\"hotel\",\"name\":\"Hotel Ned\",\"city\":\"Balmain\","
+        "\"country\":\"Australia\",\"description\":\"Undefined description for Hotel Ned\","
+        "\"features\":[\"Cable TV\",\"Toaster\",\"Microwave\"]}]";
```

Comment:
1. Recommend to simplify the data to have just a simple array (not nested). 
2. Create a MutableArray and iterate through it.
3. No need to add complexity of creating a document from a content of array.